app/models/attachment.rb

```diff
@@ -148,6 +148,25 @@ def self.pwd
     AttachmentPwd
   end
 
+  # Obtain a suitable attachment name for a recently uploaded file. If the
+  # original file name is still available, use it, otherwise, provide count-based
+  # an alternative.
+  def self.available_name(node, args={})
```

Comment:
It is a good extraction but the application's `AttachmentsController` has a copy of this method? I think the idea was to remove it from there too?